<!DOCTYPE html>



  


<html class="theme-next pisces use-motion" lang="zh-Hans">
<head><meta name="generator" content="Hexo 3.8.0">
  <meta name="google-site-verification" content="G086OtDfg3zyMNCUZ4n9po1t2K6PbweKCifJcGI8awA">
<meta charset="UTF-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
<meta name="theme-color" content="#222">

<meta name="referrer" content="no-referrer">








<meta http-equiv="Cache-Control" content="no-transform">
<meta http-equiv="Cache-Control" content="no-siteapp">
















  
  
  <link href="/lib/fancybox/source/jquery.fancybox.css?v=2.1.5" rel="stylesheet" type="text/css">







<link href="/lib/font-awesome/css/font-awesome.min.css?v=4.6.2" rel="stylesheet" type="text/css">

<link href="/css/main.css?v=5.1.4" rel="stylesheet" type="text/css">


  <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon-next.png?v=5.1.4">


  <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32-next.png?v=5.1.4">


  <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16-next.png?v=5.1.4">


  <link rel="mask-icon" href="/images/logo.svg?v=5.1.4" color="#222">





  <meta name="keywords" content="Hexo, NexT">





  <link rel="alternate" href="/atom.xml" title="暗之洋光" type="application/atom+xml">






<meta name="description" content="简介&amp;emsp;&amp;emsp;由于单细胞数据的高维度，基因长度差异，覆盖度差异及实验过程中的偏好性等因素，对前期数据进行有质量的标准化，对后续分析结果解读至关重要。标准化的分析比较多，对于常规的群体RNA分析而言，各种软件包也有相应的标准化方式，基因的定量也有RPKM ，TPM等标准化指标，单细胞数据对标准化的要求更高，本篇文章主要介绍新的标准化方法sctransform，他可以应用到Seurat中">
<meta property="og:type" content="article">
<meta property="og:title" content="换个姿势标准化单细胞数据">
<meta property="og:url" content="http://liuyang0681.github.io/2019/05/22/换个姿势标准化单细胞数据/index.html">
<meta property="og:site_name" content="暗之洋光">
<meta property="og:description" content="简介&amp;emsp;&amp;emsp;由于单细胞数据的高维度，基因长度差异，覆盖度差异及实验过程中的偏好性等因素，对前期数据进行有质量的标准化，对后续分析结果解读至关重要。标准化的分析比较多，对于常规的群体RNA分析而言，各种软件包也有相应的标准化方式，基因的定量也有RPKM ，TPM等标准化指标，单细胞数据对标准化的要求更高，本篇文章主要介绍新的标准化方法sctransform，他可以应用到Seurat中">
<meta property="og:locale" content="zh-Hans">
<meta property="og:image" content="https://upload-images.jianshu.io/upload_images/6732709-80bee295f1189e9b.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240">
<meta property="og:image" content="https://upload-images.jianshu.io/upload_images/6732709-ca1b2046c9ffe51e.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240">
<meta property="og:image" content="https://upload-images.jianshu.io/upload_images/6732709-33251aefad3a2792.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240">
<meta property="og:image" content="https://upload-images.jianshu.io/upload_images/6732709-1103c3155b4994e4.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240">
<meta property="og:image" content="https://upload-images.jianshu.io/upload_images/6732709-dd95f9cefa2e060e.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240">
<meta property="og:updated_time" content="2019-05-22T15:28:25.683Z">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="换个姿势标准化单细胞数据">
<meta name="twitter:description" content="简介&amp;emsp;&amp;emsp;由于单细胞数据的高维度，基因长度差异，覆盖度差异及实验过程中的偏好性等因素，对前期数据进行有质量的标准化，对后续分析结果解读至关重要。标准化的分析比较多，对于常规的群体RNA分析而言，各种软件包也有相应的标准化方式，基因的定量也有RPKM ，TPM等标准化指标，单细胞数据对标准化的要求更高，本篇文章主要介绍新的标准化方法sctransform，他可以应用到Seurat中">
<meta name="twitter:image" content="https://upload-images.jianshu.io/upload_images/6732709-80bee295f1189e9b.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240">



<script type="text/javascript" id="hexo.configurations">
  var NexT = window.NexT || {};
  var CONFIG = {
    root: '/',
    scheme: 'Pisces',
    version: '5.1.4',
    sidebar: {"position":"left","display":"post","offset":12,"b2t":false,"scrollpercent":false,"onmobile":false},
    fancybox: true,
    tabs: true,
    motion: {"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}},
    duoshuo: {
      userId: '0',
      author: '博主'
    },
    algolia: {
      applicationID: '',
      apiKey: '',
      indexName: '',
      hits: {"per_page":10},
      labels: {"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}
    }
  };
</script>



  <link rel="canonical" href="http://liuyang0681.github.io/2019/05/22/换个姿势标准化单细胞数据/">





  <title>换个姿势标准化单细胞数据 | 暗之洋光</title>
  








</head>

<body itemscope itemtype="http://schema.org/WebPage" lang="zh-Hans">

  
  
    
  

  <div class="container sidebar-position-left page-post-detail">
    <div class="headband"></div>
   
      <header id="header" class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-wrapper">
  <div class="site-meta ">
    

    <div class="custom-logo-site-title">
      <a href="/" class="brand" rel="start">
        <span class="logo-line-before"><i></i></span>
        <span class="site-title">暗之洋光</span>
        <span class="logo-line-after"><i></i></span>
      </a>
    </div>
      
        <p class="site-subtitle"></p>
      
  </div>

  <div class="site-nav-toggle">
    <button>
      <span class="btn-bar"></span>
      <span class="btn-bar"></span>
      <span class="btn-bar"></span>
    </button>
  </div>
</div>

<nav class="site-nav">
  

  
    <ul id="menu" class="menu">
      
        
        <li class="menu-item menu-item-home">
          <a href="/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-question-circle"></i> <br>
            
            首页
          </a>
        </li>
      
        
        <li class="menu-item menu-item-about">
          <a href="/about" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-question-circle"></i> <br>
            
            关于
          </a>
        </li>
      
        
        <li class="menu-item menu-item-categories">
          <a href="/categories/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-th"></i> <br>
            
            分类
          </a>
        </li>
      
        
        <li class="menu-item menu-item-archives">
          <a href="/archives/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-archive"></i> <br>
            
            归档
          </a>
        </li>
      
        
        <li class="menu-item menu-item-sitemap">
          <a href="/sitemap.xml" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-sitemap"></i> <br>
            
            站点地图
          </a>
        </li>
      

      
    </ul>
  

  
</nav>



 </div>
    </header>

    <main id="main" class="main">
      <div class="main-inner">
        <div class="content-wrap">
          <div id="content" class="content">
            

  <div id="posts" class="posts-expand">
    

  

  
  
  

  <article class="post post-type-normal" itemscope itemtype="http://schema.org/Article">
  
  
  
  <div class="post-block">
    <link itemprop="mainEntityOfPage" href="http://liuyang0681.github.io/2019/05/22/换个姿势标准化单细胞数据/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="name" content="浏阳（Yang)">
      <meta itemprop="description" content>
      <meta itemprop="image" content="/images/modelface.png">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="暗之洋光">
    </span>

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">换个姿势标准化单细胞数据</h1>
        

        <div class="post-meta">
          <span class="post-time">
            
              <span class="post-meta-item-icon">
                <i class="fa fa-calendar-o"></i>
              </span>
              
                <span class="post-meta-item-text">发表于</span>
              
              <time title="创建于" itemprop="dateCreated datePublished" datetime="2019-05-22T23:27:22+08:00">
                2019-05-22
              </time>
            

            

            
          </span>

          
            <span class="post-category">
            
              <span class="post-meta-divider">|</span>
            
              <span class="post-meta-item-icon">
                <i class="fa fa-folder-o"></i>
              </span>
              
                <span class="post-meta-item-text">分类于</span>
              
              
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/生信分析/" itemprop="url" rel="index">
                    <span itemprop="name">生信分析</span>
                  </a>
                </span>

                
                
              
            </span>
          

          
            
          

          
          

          

          

          

        </div>
      </header>
    

    
    
    
    <div class="post-body" itemprop="articleBody">

      
      

      
        <h5 id="简介"><a href="#简介" class="headerlink" title="简介"></a>简介</h5><p>&emsp;&emsp;由于单细胞数据的高维度，基因长度差异，覆盖度差异及实验过程中的偏好性等因素，对前期数据进行有质量的标准化，对后续分析结果解读至关重要。标准化的分析比较多，对于常规的群体RNA分析而言，各种软件包也有相应的标准化方式，基因的定量也有RPKM ，TPM等标准化指标，单细胞数据对标准化的要求更高，本篇文章主要介绍新的标准化方法<code>sctransform</code>，他可以应用到Seurat中替换其原有的标准化函数。<a href="https://www.jianshu.com/p/152e19e70646" target="_blank" rel="noopener">seuratV3简介及实操</a>有对其相应的介绍。</p>
<h5 id="安装R包及数据导入创建对象"><a href="#安装R包及数据导入创建对象" class="headerlink" title="安装R包及数据导入创建对象"></a>安装R包及数据导入创建对象</h5><figure class="highlight stylus"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line">#打开R环境直接安装</span><br><span class="line">devtools::install_github(repo = <span class="string">'ChristophH/sctransform'</span>, ref = <span class="string">'develop'</span>)</span><br><span class="line">devtools::install_github(repo = <span class="string">'satijalab/seurat'</span>, ref = <span class="string">'release/3.0'</span>)</span><br><span class="line">#也可以直接安装SeuratV3包</span><br><span class="line">install.packages(<span class="string">"seurat"</span>)</span><br><span class="line"><span class="function"><span class="title">library</span><span class="params">(Seurat)</span></span></span><br><span class="line"><span class="function"><span class="title">library</span><span class="params">(ggplot2)</span></span></span><br><span class="line"><span class="function"><span class="title">library</span><span class="params">(sctransform)</span></span></span><br><span class="line">#测试数据在<span class="number">10</span>X官网下载</span><br><span class="line">pbmc_data &lt;- Read10X(data<span class="selector-class">.dir</span> = <span class="string">"pbmc3k/filtered_gene_bc_matrices/hg19/"</span>)</span><br><span class="line">pbmc &lt;- CreateSeuratObject(counts = pbmc_data)</span><br></pre></td></tr></table></figure>
<h5 id="细节介绍"><a href="#细节介绍" class="headerlink" title="细节介绍"></a>细节介绍</h5><p>为什么在使用sctransform时可以选择更多的PC？<br>答：</p>
<ul>
<li>在<a href="https://satijalab.org/seurat/pbmc3k_tutorial.html" target="_blank" rel="noopener">标准的Seurat工作流程中，</a>我们专注于此数据集的10PC，但我们强调结果与此参数的更高设置相似。有趣的是，我们发现在使用sctransform时，我们通常会将此参数推得更高。我们相信这是因为sctransform工作流程执行更有效的规范化，强烈消除了数据的技术效果。</li>
<li>即使在标准对数归一化之后，测序深度的变化仍然是一个混淆因素（<a href="https://www.biorxiv.org/content/biorxiv/early/2019/03/18/576827.full.pdf" target="_blank" rel="noopener">见图1</a>），这种影响可以巧妙地影响更高的PC。在sctransform中，这种影响大大减轻了（<a href="https://www.biorxiv.org/content/biorxiv/early/2019/03/18/576827.full.pdf" target="_blank" rel="noopener">见图3</a>）。这意味着更高的PC更有可能代表微妙的，但与生物学相关的异质性来源 - 因此包括它们可能会改善下游分析。</li>
<li>此外，sctransform默认返回3,000个变量功能，而不是2,000。基本原理类似，额外的变量特征不太可能由细胞间的技术差异驱动，而是可能代表更微妙的生物波动。一般来说，我们发现用sctransform产生的结果较少依赖于这些参数（实际上，当在转录组中使用所有基因时，我们获得了几乎相同的结果，尽管这确实降低了计算效率）。这可以帮助用户生成更强大的结果，此外，还可以应用具有相同参数设置的标准anlaysis管道，这些参数设置可以快速应用于新数据集：</li>
</ul>
<p>例如，以下代码在单个命令中复制完整的端到端工作流：<br><figure class="highlight mel"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">pbmc &lt;- CreateSeuratObject(pbmc_data) %&gt;% PercentageFeatureSet(pattern = <span class="string">"^MT-"</span>, col.name = <span class="string">"percent.mt"</span>) %&gt;% </span><br><span class="line">    SCTransform(vars.to.regress = <span class="string">"percent.mt"</span>) %&gt;% RunPCA() %&gt;% FindNeighbors(dims = <span class="number">1</span>:<span class="number">30</span>) %&gt;% </span><br><span class="line">    RunUMAP(dims = <span class="number">1</span>:<span class="number">30</span>) %&gt;% FindClusters()</span><br></pre></td></tr></table></figure></p>
<p>sctransform存储的归一化值在哪里？<br>答：<br> 如的<a href="https://www.biorxiv.org/content/biorxiv/early/2019/03/18/576827.full.pdf" target="_blank" rel="noopener">预印本中所述</a>，sctransform使用“正则化负二项式回归”计算scRNA-seq数据中的技术噪声模型。该模型的残差是标准化值，可以是正数或负数。给定基因中给定基因的阳性残基表明，鉴于基因在群体和细胞测序深度中的平均表达，观察到比预期更多的UMI，而负残差表明相反。<br>sctransfrom的结果存储在“SCT”测定中。您可以在<a href="http://satijalab01.nygenome.org/seurat/v3.0/multimodal_vignette.html" target="_blank" rel="noopener">插图</a>，<a href="http://satijalab01.nygenome.org/seurat/essential_commands.html#multi-assay-features" target="_blank" rel="noopener">命令备忘单</a>或<a href="https://github.com/satijalab/seurat/wiki/Assay" target="_blank" rel="noopener">开发者指南中</a>了解有关Seurat中多种化验数据和命令的更多信息。</p>
<ul>
<li><code>pbmc[[&quot;SCT&quot;]]@scale.data</code>包含残差（标准化值），并直接用作PCA的输入。请注意，这个矩阵是非稀疏的，因此如果为所有基因存储，则会占用大量内存。为了节省内存，我们通过在<code>SCTransform</code>函数调用中默认设置return.only.var.genes = TRUE来仅为变量基因存储这些值。</li>
<li>协助可视化和解释。我们还将Pearson残差转换回’更正’的UMI计数。您可以将这些解释为我们希望观察到的所有细胞都被测序到相同深度的UMI计数。如果您想确切了解我们如何做到这一点，请在<a href="https://github.com/ChristophH/sctransform/blob/master/R/denoise.R" target="_blank" rel="noopener">此处</a>查看正确的功能。</li>
<li>“更正的”UMI计数存储在<code>pbmc[[&quot;SCT&quot;]]@counts</code>。我们存储这些校正计数的日志标准化版本，这<code>pbmc[[&quot;SCT&quot;]]@data</code>对可视化非常有用。</li>
<li>您可以使用更正的对数标准化计数进行差异表达和积分。然而，原则上，直接对残差（存储在<code>scale.data</code>槽中）本身执行这些计算是最佳的。Seurat v3目前不支持此功能，但很快就会支持。<h5 id="两种标准化方式比较"><a href="#两种标准化方式比较" class="headerlink" title="两种标准化方式比较"></a>两种标准化方式比较</h5>作为参考，首先应用标准的Seurat工作流程，并使用对数标准化<figure class="highlight lisp"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">pbmc_logtransform &lt;- pbmc</span><br><span class="line">pbmc_logtransform &lt;- NormalizeData(<span class="name">pbmc_logtransform</span>, verbose = FALSE)</span><br><span class="line">pbmc_logtransform &lt;- FindVariableFeatures(<span class="name">pbmc_logtransform</span>, verbose = FALSE) </span><br><span class="line">pbmc_logtransform &lt;- ScaleData(<span class="name">pbmc_logtransform</span>, verbose = FALSE) </span><br><span class="line">pbmc_logtransform &lt;- RunPCA(<span class="name">pbmc_logtransform</span>, verbose = FALSE) </span><br><span class="line">pbmc_logtransform &lt;- RunUMAP(<span class="name">pbmc_logtransform</span>,dims = <span class="number">1</span>:<span class="number">20</span>, verbose = FALSE)</span><br></pre></td></tr></table></figure>
</li>
</ul>
<p>为了比较，现在应用sctransform规范化<br><figure class="highlight kotlin"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"># Note that <span class="keyword">this</span> single command replaces NormalizeData, ScaleData, and FindVariableFeatures.</span><br><span class="line"># Transformed <span class="keyword">data</span> will be available <span class="keyword">in</span> the SCT assay, which <span class="keyword">is</span> <span class="keyword">set</span> <span class="keyword">as</span> the <span class="keyword">default</span> after running sctransform</span><br><span class="line">pbmc &lt;- SCTransform(<span class="keyword">object</span> = pbmc, verbose = FALSE)</span><br></pre></td></tr></table></figure></p>
<p>通过PCA和UMAP嵌入执行降维<br><figure class="highlight jboss-cli"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># These are now standard steps in the Seurat workflow for visualization and clustering</span></span><br><span class="line">pbmc &lt;- RunPCA<span class="params">(<span class="attr">object</span> = pbmc, <span class="attr">verbose</span> = FALSE)</span></span><br><span class="line">pbmc &lt;- RunUMAP<span class="params">(<span class="attr">object</span> = pbmc, <span class="attr">dims</span> = 1:20, <span class="attr">verbose</span> = FALSE)</span></span><br><span class="line">pbmc &lt;- FindNeighbors<span class="params">(<span class="attr">object</span> = pbmc, <span class="attr">dims</span> = 1:20, <span class="attr">verbose</span> = FALSE)</span></span><br><span class="line">pbmc &lt;- FindClusters<span class="params">(<span class="attr">object</span> = pbmc, <span class="attr">verbose</span> = FALSE)</span></span><br></pre></td></tr></table></figure></p>
<p>在sctransform和log-normalized嵌入上可视化聚类结果。<br><figure class="highlight stylus"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">pbmc_logtransform<span class="variable">$clusterID</span> &lt;- Idents(pbmc)</span><br><span class="line"><span class="function"><span class="title">Idents</span><span class="params">(pbmc_logtransform)</span></span> &lt;- <span class="string">'clusterID'</span></span><br><span class="line">plot1 &lt;- DimPlot(<span class="selector-tag">object</span> = pbmc, <span class="selector-tag">label</span> = TRUE) + NoLegend() + ggtitle(<span class="string">'sctransform'</span>) </span><br><span class="line">plot2 &lt;- DimPlot(<span class="selector-tag">object</span> = pbmc_logtransform, <span class="selector-tag">label</span> = TRUE)</span><br><span class="line">plot2 &lt;- plot2 + NoLegend() + ggtitle(<span class="string">'Log-normalization'</span>) </span><br><span class="line"><span class="function"><span class="title">CombinePlots</span><span class="params">(list(plot1,plot2)</span></span>)</span><br></pre></td></tr></table></figure></p>
<p><img src="https://upload-images.jianshu.io/upload_images/6732709-80bee295f1189e9b.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt><br>用户可以基于规范标记单独注释群集。然而，与对数归一化分析相比，sctransform归一化显示出更明显的生物学区别。例如，注意在对数归一化分析中如何将簇0,1,9和11（所有不同的T细胞簇）混合在一起。sctransform分析显示：</p>
<ul>
<li>基于CD8A，GZMK，CCL5，GZMK表达，清楚分离三个CD8 T细胞簇（幼稚，记忆，效应器）</li>
<li>基于S100A4，CCR7，IL32和ISG15清楚分离三个CD4 T细胞簇（幼稚，记忆，IFN激活）</li>
<li>B细胞群中的附加发育子结构，基于TCL1A，FCER2</li>
<li>基于XCL1和FCGR3A的NK细胞簇内的额外亚结构（CD56dim对比亮）<br>将规范标记基因可视化为小提琴图。<figure class="highlight nix"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># These are now standard steps in the Seurat workflow for visualization and clustering Visualize</span></span><br><span class="line"><span class="comment"># canonical marker genes as violin plots.</span></span><br><span class="line">VlnPlot(pbmc, <span class="attr">features</span> = c(<span class="string">"CD8A"</span>, <span class="string">"GZMK"</span>, <span class="string">"CCL5"</span>, <span class="string">"S100A4"</span>, <span class="string">"ANXA1"</span>, <span class="string">"CCR7"</span>, <span class="string">"ISG15"</span>, <span class="string">"CD3D"</span>), </span><br><span class="line">    pt.<span class="attr">size</span> = <span class="number">0.2</span>, <span class="attr">ncol</span> = <span class="number">4</span>)</span><br></pre></td></tr></table></figure>
</li>
</ul>
<p><img src="https://upload-images.jianshu.io/upload_images/6732709-ca1b2046c9ffe51e.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt><br>在sctransform嵌入上可视化规范标记基因。<br><figure class="highlight stylus"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line"><span class="function"><span class="title">FeaturePlot</span><span class="params">(object = pbmc, features = c(<span class="string">"CD8A"</span>,<span class="string">"GZMK"</span>,<span class="string">"CCL5"</span>,<span class="string">"S100A4"</span>,<span class="string">"S100A4"</span>,<span class="string">"CCR7"</span>)</span></span>, pt<span class="selector-class">.size</span> = <span class="number">0.2</span>,ncol = <span class="number">3</span>)</span><br></pre></td></tr></table></figure></p>
<p><img src="https://upload-images.jianshu.io/upload_images/6732709-33251aefad3a2792.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt><br><figure class="highlight stylus"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line"><span class="function"><span class="title">FeaturePlot</span><span class="params">(object = pbmc, features = c(<span class="string">"CD3D"</span>,<span class="string">"ISG15"</span>,<span class="string">"TCL1A"</span>,<span class="string">"FCER2"</span>,<span class="string">"XCL1"</span>,<span class="string">"FCGR3A"</span>)</span></span>, pt<span class="selector-class">.size</span> = <span class="number">0.2</span>,ncol = <span class="number">3</span>)</span><br></pre></td></tr></table></figure></p>
<p><img src="https://upload-images.jianshu.io/upload_images/6732709-1103c3155b4994e4.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt></p>
<h5 id="参考材料"><a href="#参考材料" class="headerlink" title="参考材料:"></a>参考材料:</h5><p><a href="https://rawgit.com/ChristophH/sctransform/master/inst/doc/seurat.html" target="_blank" rel="noopener">https://rawgit.com/ChristophH/sctransform/master/inst/doc/seurat.html</a><br><a href="https://satijalab.org/seurat/v3.0/sctransform_vignette.html" target="_blank" rel="noopener">https://satijalab.org/seurat/v3.0/sctransform_vignette.html</a></p>
<p>关注微信公众号，第一时间获取更新<br><img src="https://upload-images.jianshu.io/upload_images/6732709-dd95f9cefa2e060e.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt></p>

      
    </div>
    
    
    

    

    

    

<div>
  
    <div>
    
        <div style="text-align:center;color: #ccc;font-size:14px;">-------------　　　　本文结束　<i class="fa fa-heart"></i>　感谢您的阅读　　　　-------------</div>
    
</div>

  
</div>

    <footer class="post-footer">
      

      
      
      

      
        <div class="post-nav">
          <div class="post-nav-next post-nav-item">
            
              <a href="/2019/05/21/单细胞数据整合分析哪家强/" rel="next" title="单细胞数据整合分析哪家强?">
                <i class="fa fa-chevron-left"></i> 单细胞数据整合分析哪家强?
              </a>
            
          </div>

          <span class="post-nav-divider"></span>

          <div class="post-nav-prev post-nav-item">
            
              <a href="/2019/05/23/2019年人类细胞图谱计划（HCA）日本会议/" rel="prev" title="2019年人类细胞图谱计划（HCA）日本会议">
                2019年人类细胞图谱计划（HCA）日本会议 <i class="fa fa-chevron-right"></i>
              </a>
            
          </div>
        </div>
      

      
      
    </footer>
  </div>
  
  
  
  </article>



    <div class="post-spread">
      
    </div>
  </div>


          </div>
          


          

  



        </div>
        
          
  
  <div class="sidebar-toggle">
    <div class="sidebar-toggle-line-wrap">
      <span class="sidebar-toggle-line sidebar-toggle-line-first"></span>
      <span class="sidebar-toggle-line sidebar-toggle-line-middle"></span>
      <span class="sidebar-toggle-line sidebar-toggle-line-last"></span>
    </div>
  </div>

  <aside id="sidebar" class="sidebar">
    
    <div class="sidebar-inner">

      

      
        <ul class="sidebar-nav motion-element">
          <li class="sidebar-nav-toc sidebar-nav-active" data-target="post-toc-wrap">
            文章目录
          </li>
          <li class="sidebar-nav-overview" data-target="site-overview-wrap">
            站点概览
          </li>
        </ul>
      

      <section class="site-overview-wrap sidebar-panel">
        <div class="site-overview">
          <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
            
              <img class="site-author-image" itemprop="image" src="/images/modelface.png" alt="浏阳（Yang)">
            
              <p class="site-author-name" itemprop="name">浏阳（Yang)</p>
              <p class="site-description motion-element" itemprop="description"></p>
          </div>

          <nav class="site-state motion-element">

            
              <div class="site-state-item site-state-posts">
              
                <a href="/archives/">
              
                  <span class="site-state-item-count">26</span>
                  <span class="site-state-item-name">日志</span>
                </a>
              </div>
            

            
              
              
              <div class="site-state-item site-state-categories">
                <a href="/categories/index.html">
                  <span class="site-state-item-count">6</span>
                  <span class="site-state-item-name">分类</span>
                </a>
              </div>
            

            

          </nav>

          
            <div class="feed-link motion-element">
              <a href="/atom.xml" rel="alternate">
                <i class="fa fa-rss"></i>
                RSS
              </a>
            </div>
          

          
            <div class="links-of-author motion-element">
                
                  <span class="links-of-author-item">
                    <a href="https://github.com/liuyang0681" target="_blank" title="Github">
                      
                        <i class="fa fa-fw fa-globe"></i>Github</a>
                  </span>
                
                  <span class="links-of-author-item">
                    <a href="mailto:liuyang0681@hotmail.com" target="_blank" title="Email">
                      
                        <i class="fa fa-fw fa-globe"></i>Email</a>
                  </span>
                
            </div>
          

          
          

          
          

          

        </div>
      </section>

      
      <!--noindex-->
        <section class="post-toc-wrap motion-element sidebar-panel sidebar-panel-active">
          <div class="post-toc">

            
              
            

            
              <div class="post-toc-content"><ol class="nav"><li class="nav-item nav-level-5"><a class="nav-link" href="#简介"><span class="nav-number">1.</span> <span class="nav-text">简介</span></a></li><li class="nav-item nav-level-5"><a class="nav-link" href="#安装R包及数据导入创建对象"><span class="nav-number">2.</span> <span class="nav-text">安装R包及数据导入创建对象</span></a></li><li class="nav-item nav-level-5"><a class="nav-link" href="#细节介绍"><span class="nav-number">3.</span> <span class="nav-text">细节介绍</span></a></li><li class="nav-item nav-level-5"><a class="nav-link" href="#两种标准化方式比较"><span class="nav-number">4.</span> <span class="nav-text">两种标准化方式比较</span></a></li><li class="nav-item nav-level-5"><a class="nav-link" href="#参考材料"><span class="nav-number">5.</span> <span class="nav-text">参考材料:</span></a></li></ol></div>
            

          </div>
        </section>
      <!--/noindex-->
      

      

    </div>
  </aside>


        
      </div>
    </main>

    <footer id="footer" class="footer">
      <div class="footer-inner">
        <div class="powered-by">
<i class="fa fa-user-md"></i><span id="busuanzi_container_site_uv">
  本站访客数:<span id="busuanzi_value_site_uv"></span>
</span>
</div>

<script async src="https://dn-lbstatics.qbox.me/busuanzi/2.3/busuanzi.pure.mini.js"></script>
<div class="copyright">&copy; <span itemprop="copyrightYear">2019</span>
  <span class="with-love">
    <i class="fa fa-user"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">浏阳（Yang)</span>

  
</div>


  <div class="powered-by">由 <a class="theme-link" target="_blank" href="https://hexo.io">Hexo</a> 强力驱动</div>



  <span class="post-meta-divider">|</span>



  <div class="theme-info">主题 &mdash; <a class="theme-link" target="_blank" href="https://github.com/iissnan/hexo-theme-next">NexT.Pisces</a> v5.1.4</div>




        







        
      </div>
    </footer>

    
      <div class="back-to-top">
        <i class="fa fa-arrow-up"></i>
        
      </div>
    

    

  </div>

  

<script type="text/javascript">
  if (Object.prototype.toString.call(window.Promise) !== '[object Function]') {
    window.Promise = null;
  }
</script>









  


  











  
  
    <script type="text/javascript" src="/lib/jquery/index.js?v=2.1.3"></script>
  

  
  
    <script type="text/javascript" src="/lib/fastclick/lib/fastclick.min.js?v=1.0.6"></script>
  

  
  
    <script type="text/javascript" src="/lib/jquery_lazyload/jquery.lazyload.js?v=1.9.7"></script>
  

  
  
    <script type="text/javascript" src="/lib/velocity/velocity.min.js?v=1.2.1"></script>
  

  
  
    <script type="text/javascript" src="/lib/velocity/velocity.ui.min.js?v=1.2.1"></script>
  

  
  
    <script type="text/javascript" src="/lib/fancybox/source/jquery.fancybox.pack.js?v=2.1.5"></script>
  

  
  
    <script type="text/javascript" src="/lib/canvas-nest/canvas-nest.min.js"></script>
  


  


  <script type="text/javascript" src="/js/src/utils.js?v=5.1.4"></script>

  <script type="text/javascript" src="/js/src/motion.js?v=5.1.4"></script>



  
  


  <script type="text/javascript" src="/js/src/affix.js?v=5.1.4"></script>

  <script type="text/javascript" src="/js/src/schemes/pisces.js?v=5.1.4"></script>



  
  <script type="text/javascript" src="/js/src/scrollspy.js?v=5.1.4"></script>
<script type="text/javascript" src="/js/src/post-details.js?v=5.1.4"></script>



  


  <script type="text/javascript" src="/js/src/bootstrap.js?v=5.1.4"></script>



  


  




	





  





  












  





  

  

  

  
  

  

  

  

</body>
</html>
